Switch suricata to repeat mode

filippocarletti opened the pull request:
https://github.com/NethServer/dev/issues/6205

The exclusion for 127.0.0.1/32 in shorewall/rules may be required only on non-gateways (i.e. systems having only green zones), more tests needed.

See also NethServer/nethserver-firewall-base#143

filippocarletti committed:
Remove suppression for Hylafax false positive

Sid 2011124 in now disabled by default

filippocarletti committed:
Remove nfqueue

filippocarletti committed:
Suricata nfq repeat mode

Using repeat mode we can also enable fail-open because suricata verdict
is non-terminal (packets will be dropped or sent to iptables)

gsanchietti commented on root/etc/e-smith/templates/etc/shorewall/rules/15IPS:
```diff
@@ -0,0 +1,6 @@
+{
+    my $status = $suricata{'status'} || 'enabled';
+    if ($status eq 'enabled') {
+        $OUT.="\n?SECTION ALL\nNFQUEUE(bypass) all+ all+:!127.0.0.1/32 - - - - - - !0x10/0x10\n\n?SECTION NEW\n";
```
Firewall sections must be handled inside firewall template.

Also the exclusion should be more like `127.0.0.0/24`.
Moreover, why not exclude all green IP addresses?

gsanchietti commented on root/etc/e-smith/templates/etc/suricata/suricata.yaml/10base:
```diff
@@ -1378,12 +1378,12 @@ profiling:
-#  repeat-mark: 1
-#  repeat-mask: 1
+  mode: repeat
+  repeat-mark: 16
+  repeat-mask: 16
```
Is this mark good even if ndpi, qos and multiwan are enabled?

filippocarletti replied:
Yes

gsanchietti reviewed (commented):
Also make sure to bind this version of `nethserver-suricata` with a specific version of `nethserver-firewall-base` otherwise it will break current installations.

filippocarletti committed:
Improve localhost traffic bypass

Do not send to suricata traffic moving inside localhost, explicitly
ACCEPTing traffic to/from the firewall itself (instead of excluding
traffic to 127.0.0.1).

Example of double suricata alert without exclusion:

06/13/2020-10:36:22.781714  [**] [1:2027863:2] ET INFO Observed DNS Query to .biz TLD [**] [Classification: Potentially Bad Traffic] [Priority: 2] {UDP} 127.0.0.1:50292 -> 127.0.0.1:53
06/13/2020-10:36:22.782175  [**] [1:2027863:2] ET INFO Observed DNS Query to .biz TLD [**] [Classification: Potentially Bad Traffic] [Priority: 2] {UDP} 192.168.122.35:49359 -> 192.168.122.1:53

filippocarletti committed:
Require correct nethserver-firewall-base version

filippocarletti commented:
> Also make sure to bind this version of `nethserver-suricata` with a specific version of `nethserver-firewall-base` otherwise it will break current installations.

Done:
`+Requires: nethserver-firewall-base > 3.9.2`

gsanchietti reviewed (approved):
Wait for the issue before merge.
